src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -142,131 +147,239 @@ class CrossDeviceLinkUI extends Component {
+        ) : (
+          <PageTitle
+            title={translate('cross_device.link.title')}
+            subTitle={translate(`cross_device.link.${currentViewId}_sub_title`)} />
+        )}
+        <div className={classNames(theme.thickWrapper, style.secureLinkView)}>
+          <div id="selectedLinkView" role="region" aria-live="polite">
+            {currentView.render()}
           </div>
-          <div className={style.smsSection}>
-            <div className={style.label}>{translate('cross_device.link.sms_label')}</div>
-            <div className={style.numberInputSection}>
-              <div className={classNames(style.inputContainer, {[style.fieldError]: invalidNumber})}>
-                <PhoneNumberInputLazy { ...this.props} clearErrors={this.clearErrors} />
-              </div>
-              <Button
-                ariaLive="polite"
-                ariaBusy={this.state.sending}
-                className={classNames(style.btn, {[style.sending]: this.state.sending})}
-                variants={["primary"]}
-                onClick={this.handleSendLinkClick}
-                disabled={this.state.sending || invalidNumber}
-              >
-                {buttonCopy}
-              </Button>
-            </div>
-          </div>
-          <div role="alert" aria-atomic="true">
-            { invalidNumber && <div className={style.numberError}>{translate('errors.invalid_number.message')}</div> }
-          </div>
-          <div className={style.copyLinkSection}>
-            <div tabIndex="0" className={style.label}>{translate('cross_device.link.copy_link_label')}</div>
-            <div className={classNames(style.linkContainer, this.state.copySuccess && style.copySuccess)}>
-              <span className={style.linkText} ref={(element) => this.linkText = element}>
-                {mobileUrl}
-              </span>
-              { document.queryCommandSupported('copy') &&
-                <div className={style.actionContainer} aria-live="polite">
-                  <button
-                    type="button"
-                    onClick={() => copyToClipboard(mobileUrl, this.onCopySuccess)}
-                    className={style.copyToClipboard}
-                  >
-                    {linkCopy}
-                  </button>
-                </div>
-              }
-            </div>
-            <hr className={style.divider} />
+          <p className={style.styledLabel}>
+            {translate('cross_device.link.options_divider_label')}
+          </p>
+          <div className={style.viewOptions} aria-controls="selectedLinkView">
+            {secureLinkViews
+              .filter(view => view.id !== this.state.currentViewId)
+              .map(view => (
+                <button
+                  type="button"
+                  className={classNames(theme.link, style.viewOption, style[view.className])}
+                  onClick={(evt) => {
+                    evt.target.blur()
```

Comment:
Why do you need to blur here?

Reply:
Not needed actually. Another thing I did not clean up after fixing screenreader focus/order.